src/libcore/mem/manually_drop.rs

```diff
@@ -113,19 +128,26 @@ impl<T> ManuallyDrop<T> {
 }
 
 impl<T: ?Sized> ManuallyDrop<T> {
-    /// Manually drops the contained value.
+    /// Manually drops the contained value. This is exactly equivalent to calling
+    /// [`ptr::drop_in_place`] with a pointer to the contained value. As such, unless
+    /// the contained value is a packed struct, the destructor will be called in-place
+    /// without moving the value, and this can be used to safely drop pinned data.
```

Comment:
Could you make sure that we have a similar comment in the `drop_in_place` docs? Currently we just have it in the "examples", but that's a bit late for my taste. Maybe after

> Unaligned values cannot be dropped in place, they must be copied to an aligned location first using ptr::read_unaligned.

it can say

> For packed structs, this move is done automatically by the compiler. This means the fields of packed structs
> are *not* dropped in-place.